src/main/java/org/elasticsearch/index/mapper/core/CompletionFieldMapper.java

```diff
@@ -166,16 +168,13 @@ public CompletionFieldMapper build(Mapper.BuilderContext context) {
                 if (fieldName.equals("type")) {
                     continue;
                 }
-                if (fieldName.equals("analyzer")) {
-                    NamedAnalyzer analyzer = getNamedAnalyzer(parserContext, fieldNode.toString());
-                    builder.indexAnalyzer(analyzer);
-                    builder.searchAnalyzer(analyzer);
-                    iterator.remove();
-                } else if (Fields.INDEX_ANALYZER.match(fieldName)) {
-                    builder.indexAnalyzer(getNamedAnalyzer(parserContext, fieldNode.toString()));
+                if (Fields.ANALYZER.equals(fieldName) ||
+                    // for backcompat, reading old indexes, remove for v3.0
+                    fieldName.equals("index_analyzer") && parserContext.indexVersionCreated().before(Version.V_2_0_0)) {
```

Comment:
I was a bit confused that this is part of the if statement yet has the same indentation level that the block of code below. Maybe add 2 or 4 spaces of indentation here?